idl/chimera.pro

```diff
@@ -0,0 +1,535 @@
+r = ind[0].r_sun
+w = where((xgrid-center[0])^2+(ygrid-center[1])^2 lt r^2 and dat0 lt 4000 and dat1 lt 4000 and dat2 lt 4000)
+
+;=====create intensity ratio arrays=============
+for i =0L, n_elements(w)-1 do begin
+	tem[dat0[w[i]],dat1[w[i]]]=tem[dat0[w[i]],dat1[w[i]]]+1
+	tmp[dat0[w[i]],dat2[w[i]]]=tmp[dat0[w[i]],dat2[w[i]]]+1
+	tep[dat1[w[i]],dat2[w[i]]]=tep[dat1[w[i]],dat2[w[i]]]+1
+endfor
+
+;============make a multi-wavelength image for contours==================
+truecolorimage=bytarr(s[1],s[2],3)
+truecolorimage[*,*,2]=bytscl(alog10(map[0].data),max=3.9,min=1.2)
+truecolorimage[*,*,1]=bytscl(alog10(map[1].data),max=3.0,min=1.4)
+truecolorimage[*,*,0]=bytscl(alog10(map[2].data),max=2.7,min=0.8)
+
+t0=truecolorimage[*,*,0]
+t1=truecolorimage[*,*,1]
+t2=truecolorimage[*,*,2]
+
+;====create 3 segmented bitmasks bitmasks=====
+msk[where(float(t2)/t0 ge ((mean(dat0)*0.6357)/(mean(dat2))))]=1
+mak[where(float(t0)+t1 lt (0.7*(mean(dat1)+mean(dat2))))]=1
+mas[where(float(t2)/t1 ge ((mean(dat0)*1.5102)/(mean(dat1))))]=1
+
+;====logical conjunction of 3 segmentations=======
+def=msk*mak*mas
+
+;====plot tricolour image with lon/lat conotours=======
+ax=indgen(s[1])
+ay=indgen(s[2])
+!p.multi=[0,1,1]
+plot_image,truecolorimage,title='CHIMERA Coronal Holes at '+map[1].time+' UT',xstyle=5,ystyle=5,charsize=2,charthick=3
+Contour,lon,ax,ay,/over,levels=[-90,-80,-70,-60,-50,-40,-30,-20,-10,0,10,20,30,40,50,60,70,80,90],color='FFFFFF'xL,c_linestyle=1
+Contour,lat,ax,ay,/over,levels=[-90,-80,-70,-60,-50,-40,-30,-20,-10,0,10,20,30,40,50,60,70,80,90],color='FFFFFF'xL,c_linestyle=1
+xx=0
+yy=0
+axis,xx,xtitle='X (arcsecs)',xaxis=0,xrange=[-(s[1]/2)*dattoarc,(s[1]/2)*dattoarc],xstyle=1,charsize=2,charthick=3
+axis,xx,yy,xtitle='X (arcsecs)',xaxis=0,xrange=[-(s[1]/2)*dattoarc,(s[1]/2)*dattoarc],xstyle=1,yaxis=0,yrange=[-(s[2]/2)*dattoarc,(s[2]/2)*dattoarc],ystyle=1,charsize=2,charthick=3,ytitle='Y (arcsecs)'
+xx=s[1]
+yy=s[2]
+axis,xaxis=1,xrange=[-(s[1]/2)*dattoarc,(s[1]/2)*dattoarc],xstyle=1,xtickname=[' ',' ',' ',' ',' ']
+axis,yy,yaxis=1,yrange=[-(s[1]/2)*dattoarc,(s[1]/2)*dattoarc],ystyle=1,ytickname=[' ',' ',' ',' ',' ']
+XYOUTS,0.12,0.125,'SDO/AIA 0211A '+ind[2].date_obs,color='0000FF'xL,/NORMAL
+XYOUTS,0.12,0.11,'SDO/AIA 0193A '+ind[1].date_obs,color='00FF00'xL,/NORMAL
+XYOUTS,0.12,0.095,'SDO/AIA 0171A '+ind[0].date_obs,color='FF0000'xL,/NORMAL
+;void = cgSnapshot(File=outpath+'/pngs/saia/saia_171_193_211_fd_'+time2file(map[1].time,/seconds), /PNG, /NoDialog)
+
+;======removes off detector mis-identifications==========
+circ[*]=0
+r = (s[1]/2.0)-100
+xgrid = (fltarr(s[2])+1)##indgen(s[1])
+ygrid = indgen(s[2])##(fltarr(s[1])+1)
+center = [fix(s[1]/2.),fix(s[2]/2.)]
+w = where((xgrid-center[0])^2+(ygrid-center[1])^2 le r^2)
+circ[w] = 1.0
+def=def*circ
+
+;=======Seperates on-disk and off-limb CHs===============
+circ[*]=0
+r = (rs/dattoarc)-10
+w = where((xgrid-center[0])^2+(ygrid-center[1])^2 le r^2)
+circ[w] = 1.0
```

Comment:
I don't know what you mean by this but I will combine all these removals into 'circ' based processes into a single process